### ISSUE TITLE
Why are P, R, and mAP all 0 during yolov5 training

### ISSUE THREAD

**Issue by Yzs007** [ultralytics/yolov5#12373]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

 1.  parser.add_argument('--weights', type=str, default=‘’, help='initial weights path')  
        Do not use pre trained weights
 
      Epoch    GPU_mem   box_loss   obj_loss   cls_loss  Instances       Size
      0/199      3.65G     0.1099     0.0724     0.1053        195        640: 100%|██████████| 8/8 00:05
                 Class     Images  Instances          P          R      mAP50   mAP50-95: 100%|██████████| 4/4 00:00
                   all        128        929          0          0          0          0

3.   parser.add_argument('--weights', type=str, default=ROOT / 'yolov5s.pt', help='initial weights path')  
      Use pre trained weights

Epoch    GPU_mem   box_loss   obj_loss   cls_loss  Instances       Size
      0/199      3.65G    0.04544    0.06533    0.01638        195        640: 100%|██████████| 8/8 00:05
                 Class     Images  Instances          P          R      mAP50   mAP50-95: 100%|██████████| 4/4 00:00
                   all        128        929      0.671      0.606       0.68      0.447

**Comment by glenn-jocher** [comment]
The reason for P, R, and mAP values being 0 in your training results is likely that you are not using pre-trained weights. Training from scratch without pre-trained weights requires more epochs to converge. You should also ensure that your dataset is properly labeled and the training configuration is appropriate for your specific use case. If you switch to using pre-trained weights, as in your second case, you can see that the P, R, and mAP values are non-zero. This is expected, as pre-trained weights provide a good initialization for the model and facilitate training convergence.

**Comment by glenn-jocher** [comment]
@Yzs007 it seems that when you use the pre-trained weights, you achieve significantly better precision, recall, and mAP, which is expected. Using pre-trained weights helps the model with initializing better feature representations, and hence should typically result in better performance. Let me know if you have any additional questions.

**Comment by glenn-jocher** [comment]
@Yzs007 the issue is likely due to training an initially untrained model from scratch rather than using pretrained weights. When training without pretrained weights, the model starts with random parameters and needs many iterations to converge. Using pretrained weights will usually produce better initial results. The mAP of 0 indicates that the model is not producing any meaningful detections. Try using pretrained weights and ensuring that your dataset and labels are correctly formatted.
